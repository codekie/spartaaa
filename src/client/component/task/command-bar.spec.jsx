import React from 'react';
import CommandBar from './command-bar.jsx';
import { shallow } from 'enzyme';

describe('renders correctly', () => {
    let _context = null;

    beforeEach(() => { _context = _createContext(); });

    it('inactive task', () => {
        expect(_context.createElement()).toMatchSnapshot();
    });
    it('active task', () => {
        expect(_context.createElement({ isActive: true })).toMatchSnapshot();
    });
});

describe('click handlers', () => {
    let _context = null;

    beforeEach(() => { _context = _createContext(); });

    it('should call `activate` on click on the activation-button', () => {
        const tree = _context.createElement({ isActive: true });
        tree.find('.btn-activation').simulate('click');
        expect(_context.activateTask.mock.calls.length).toEqual(0);
        expect(_context.deactivateTask.mock.calls.length).toEqual(1);
    });
    it('should call `deactivate` on click on the activation-button', () => {
        const tree = _context.createElement({ isActive: false });
        tree.find('.btn-activation').simulate('click');
        expect(_context.activateTask.mock.calls.length).toEqual(1);
        expect(_context.deactivateTask.mock.calls.length).toEqual(0);
    });
});


function _createContext() {
    const activateTask = jest.fn(),
        deactivateTask = jest.fn(),
        toggleNext = jest.fn();
    return {
        createElement: ({ isActive = false } = {}) => _createElement({ activateTask, deactivateTask, toggleNext,
            isActive
        }),
        activateTask,
        deactivateTask
    };
}

function _createElement({ activateTask, deactivateTask, toggleNext, isActive }) {
    return shallow(<CommandBar activateTask={activateTask} deactivateTask={deactivateTask} taskId={1}
        isActive={isActive} tags={['next']} toggleNext={toggleNext} />);
}
