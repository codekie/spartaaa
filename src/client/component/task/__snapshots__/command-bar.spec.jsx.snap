// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly active task 1`] = `
<Panel
  className="task-command-bar"
  renderAs="nav"
>
  <Button
    className="is-grouped btn-activation"
    color={null}
    disabled={false}
    fullwidth={false}
    inverted={false}
    isStatic={false}
    loading={false}
    onClick={[Function]}
    outlined={false}
    remove={false}
    renderAs="button"
    reset={false}
    rounded={false}
    size={null}
    state={null}
    style={Object {}}
    submit={false}
    text={false}
  >
    <Icon
      align={null}
      className="is-small"
      color={null}
      icon={null}
      size={null}
      style={Object {}}
    >
      <FontAwesomeIcon
        border={false}
        className="task-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f04d",
              "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z",
            ],
            "iconName": "stop",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Icon>
  </Button>
</Panel>
`;

exports[`renders correctly inactive task 1`] = `
<Panel
  className="task-command-bar"
  renderAs="nav"
>
  <Button
    className="is-grouped btn-activation"
    color={null}
    disabled={false}
    fullwidth={false}
    inverted={false}
    isStatic={false}
    loading={false}
    onClick={[Function]}
    outlined={false}
    remove={false}
    renderAs="button"
    reset={false}
    rounded={false}
    size={null}
    state={null}
    style={Object {}}
    submit={false}
    text={false}
  >
    <Icon
      align={null}
      className="is-small"
      color={null}
      icon={null}
      size={null}
      style={Object {}}
    >
      <FontAwesomeIcon
        border={false}
        className="task-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f04b",
              "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
            ],
            "iconName": "play",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Icon>
  </Button>
</Panel>
`;
